### DISCUSSION TITLE
Getting back data from a Mysql custom query

### DISCUSSION THREAD

**Discussion by gmentzik** [Budibase/budibase#8129] [Help] [answered]
Hi all,

The only way I have found to get a field value from a custom Mysql query selecting one table row and executed as a button action is using a binding like:
`{{ query1.result.data.0.requested_field }}`, where 0 is the 1st element in data array and requested_field is the (column/field) name.

Is there any other cleverer way? I cannot find information on the topic.

**Comment by melohagan** [comment, marked as answer]
Hey @gmentzik 

As per the docs that is spot on: https://docs.budibase.com/docs/chaining-actions#action-referencing

There isn't really a neater way than that because custom queries always return an array of rows, even if it is just one row.

**Reply by gmentzik** [reply]
Thank you!

**Comment by ghost** [comment]
Just for help, the link above return a 404 error.
Here the new one : https://docs.budibase.com/docs/chaining-actions#action-referencing

Tips : the binding is CAPS sensitive ! ... I passed 2 hours to search why my reference doesn't match ...

**Reply by melohagan** [reply]
Thanks @Jules-seluJ Updated my link in the original answer as well.
